exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -147,27 +163,107 @@ func validateAndSanitizeExternalLabels(cfg *Config) (map[string]string, error) {
+
+		// Calculate the hash value based on the signature.
+		hash := crc32.ChecksumIEEE([]byte(signature))
+
+		// Convert the hash value to an integer.
+		hashInt := int(hash)
+
+		// Determine the array index based on the hash value.
+		index := hashInt % len(partitionedTS)
+
+		// Append the time series to the corresponding array.
+		partitionedTS[index] = append(partitionedTS[index], ts)
+	}
+
+	return partitionedTS
+}
+
 func (prwe *prwExporter) handleExport(ctx context.Context, tsMap map[string]*prompb.TimeSeries) error {
 	// There are no metrics to export, so return.
 	if len(tsMap) == 0 {
 		return nil
 	}
 
-	// Calls the helper function to convert and batch the TsMap to the desired format
-	requests, err := batchTimeSeries(tsMap, maxBatchByteSize)
-	if err != nil {
-		return err
+	partitionedTS := partitionTimeSeries(tsMap)
+
+	// Create channels for each partitioned array.
+	channels := make([]chan []*prompb.TimeSeries, len(partitionedTS))
+	for i := range channels {
+		channels[i] = make(chan []*prompb.TimeSeries)
 	}
-	if !prwe.walEnabled() {
-		// Perform a direct export otherwise.
-		return prwe.export(ctx, requests)
+
+	// Create a worker goroutine for each channel.
+	for i, tsArray := range partitionedTS {
+		prwe.wg.Add(1)
+		go prwe.exportWorker(ctx, channels[i])
+		channels[i] <- tsArray // Send the partitioned array to the corresponding channel
+		close(channels[i])    // Close the channel after sending the data
 	}
 
-	// Otherwise the WAL is enabled, and just persist the requests to the WAL
-	// and they'll be exported in another goroutine to the RemoteWrite endpoint.
-	if err = prwe.wal.persistToWAL(requests); err != nil {
-		return consumererror.NewPermanent(err)
+	// Wait for all workers to finish
+	for _, ch := range channels {
+		<-ch
+	}
+
+	return nil
+}
+
+// exportWorker is the worker function that consumes a channel of partitioned time series and performs the export.
+func (prwe *prwExporter) exportWorker(ctx context.Context, tsArrayChan <-chan []*prompb.TimeSeries) error {
+	for tsArray := range tsArrayChan {
+		// Create a new map for the partitioned time series.
+		partitionedMap := make(map[string]*prompb.TimeSeries)
+		for _, ts := range tsArray {
+			partitionedMap[ts.String()] = ts
+		}
+
+		// Calls the helper function to convert and batch the partitioned TsMap to the desired format
+		requests, err := batchTimeSeries(partitionedMap, maxBatchByteSize)
+		if err != nil {
+			prwe.wg.Done()
+			return err
+		} else if !prwe.walEnabled() {
+			// Perform a direct export otherwise.
+			if err := prwe.export(ctx, requests); err != nil {
+				prwe.wg.Done()
+				return err
+			}
+		} else {
+			// Otherwise, the WAL is enabled, and just persist the requests to the WAL
+			// and they'll be exported in another goroutine to the RemoteWrite endpoint.
+			if err := prwe.wal.persistToWAL(requests); err != nil {
+				prwe.wg.Done()
+				return err
+			}
+		}
 	}
+
+	prwe.wg.Done()
 	return nil
```

Comment:
after you partition the time series, they have to be submitted to the backend sequentially in each worker. I can see here that you are still calling the `export` method, which will then submit each partition in parallel.....